controllers/util/predicates.go

```diff
@@ -68,3 +70,36 @@ func (p MetadataChangedPredicate) Update(e event.UpdateEvent) bool {
 
 	return metaChanged
 }
+
+// Name Predicate return a predicate the filter events produced
+// by resources that matches the given name
+func NamePredicate(name string) predicate.Predicate {
+	return predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		return obj.GetName() == name
+	})
+}
+
+func CrdCreateAndDeletePredicate(log *logr.Logger, AvailableCrds map[string]bool, CrdName string) predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(_ event.TypedCreateEvent[client.Object]) bool {
```

Comment:
Why use the generic variant and `[client.Object]`, that does not make any sense. 
Either use the non-generic variant where there is an underlying assumption that type os `client.Object` or use a proper type (in this case is the ObjectMetadata type)